### PR TITLE
HeaderCommentFixer - throw exception on invalid header configuration

### PR DESCRIPTION
**Pull request by SpacePossum**
closes https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/4075

### CODE REVIEW

**Comment by kubawerlos** [comment]
What about `comment` type? It also cannot have `*/`.

**Comment by SpacePossum** [comment]
thanks @kubawerlos , updated :+1:

File: src/Fixer/Comment/HeaderCommentFixer.php [hunk 1]
```diff
@@ -145,14 +146,20 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens)
...
             (new FixerOptionBuilder('header', 'Proper header content.'))
                 ->setAllowedTypes(['string'])
-                ->setNormalizer(static function (Options $options, $value) {
+                ->setNormalizer(static function (Options $options, $value) use ($fixerName) {
                     if ('' === trim($value)) {
                         return '';
                     }
 
+                    if (false !== strpos($value, '*/')) {
+                        throw new InvalidFixerConfigurationException($fixerName, 'Cannot use \'*/\' in header.');
```
Comment (kubawerlos) [inline src/Fixer/Comment/HeaderCommentFixer.php]: Can't `$this->getName()` be instead of `$fixerName` here?

Comment (dmvdbrugge) [reply src/Fixer/Comment/HeaderCommentFixer.php]: I first thought the same, however, this is a static closure, thus not bound to `$this`. See http://php.net/manual/en/functions.anonymous.php#functions.anonymous-functions.static

Comment (kubawerlos) [reply src/Fixer/Comment/HeaderCommentFixer.php]: Do we need this closure to be static?

**Pull request merged**